packages/squiggle-lang/src/value/boxed.ts

```diff
@@ -0,0 +1,62 @@
+import { ImmutableMap } from "../utility/immutableMap.js";
+import { Value, vString } from "./index.js";
+
+export type BoxedArgsType = {
+  name?: string;
+  description?: string;
+  showAs?: Value;
+};
+
+//I expect these to get much more complicated later, so it seemed prudent to make a class now.
+export class BoxedArgs {
```

Comment:
I feel kind of weird about this class, but it seemed to work well enough in practice.